mne/stats/regression.py

```diff
@@ -244,19 +248,28 @@ def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
            considerations. Psychophysiology, 52(2), 169-189.
     """
     if isinstance(solver, string_types):
+        if solver not in {"cholesky"}:
+            raise ValueError("No such solver: {0}".format(solver))
         if solver == 'cholesky':
             def solver(X, y):
                 a = (X.T * X).toarray()  # dot product of sparse matrices
-                return linalg.solve(a, X.T * y.T, sym_pos=True,
+                return linalg.solve(a, X.T * y, sym_pos=True,
                                     overwrite_a=True, overwrite_b=True).T
-
-        else:
-            raise ValueError("No such solver: {0}".format(solver))
+    elif callable(solver):
```

Comment:
You could always supply a callable (this is why you raised the issue in the first place :)), I'm just adding the warning for the fix.